pilot/pkg/features/ambient.go

```diff
@@ -44,6 +44,11 @@ var (
 		"PILOT_PREFER_SENDING_HBONE",
 		false, false,
 		"If enabled, HBONE will be preferred when sending to destinations. ")
+
+	DefaultAllowFromWaypoint = registerAmbient(
+		"PILOT_DEFAULT_ENFORCE_FROM_WAYPOINT",
+		true, false,
+		"If enabled, zTunnels will enforce that incoming traffic has traversed a Waypoint if traffic to an endpoint may traverse one.")
```

Comment:
s/enforce/allow/ 

And clarify what 'may traverse' - anything may traverse a waypoint. Do we expect a special setting for a workload to allow a particular waypoint ? Are all waypoints allowed ( I don't think the design requires waypoints to be per namespace only )

Reply:
Good clarifications. 

Allow is definitely more correct. Compared to 0 policy, this does start to enforce that the only the Waypoint's SA can send traffic here.

"may traverse" would actually be "may be redirected by a zTunnel" - meaning "whether the workload is part of a service that is marked to use the waypoint OR the workload itself is marked to use the waypoint".